src/itoa_ljust.cpp

```diff
@@ -120,7 +120,7 @@ namespace itoa_ljust {
         uint32_t u = i;
         if (i < 0) {
             *p++ = '-';
-            u = -u;
+            u = (uint32_t)(-(int32_t)u);
```

Comment:
Unary minus applied to a signed integer is undefined behavior for targets that use two's complement when the value is the most negative value (INT_MIN) representable in two's complement, it causes signed overflow.   The compiler can assume this never happens.  On clang/llvm this doesn't seem to cause a problem, but on gcc with -O3 optimization the compiler takes advantage of this assumption and generates code that gives the wrong answer (or causes segmentation fault).